code/modules/clothing/spacesuits/rig.dm

```diff
@@ -97,639 +111,310 @@
 	allowed = list(/obj/item/device/flashlight,/obj/item/weapon/tank,/obj/item/weapon/storage/bag/ore,/obj/item/device/t_scanner,/obj/item/weapon/pickaxe, /obj/item/device/rcd, /obj/item/weapon/wrench/socket)
 	max_heat_protection_temperature = SPACE_SUIT_MAX_HEAT_PROTECTION_TEMPERATURE
 	pressure_resistance = 200 * ONE_ATMOSPHERE
+	actions_types = list(/datum/action/item_action/toggle_rig_suit)
+
+	var/activated = FALSE
+	var/list/modules = list()
+	var/list/initial_modules = list(/obj/item/rig_module/health_readout)
+
 	var/obj/item/clothing/head/helmet/space/rig/H = null
-	var/head_type = /obj/item/clothing/head/helmet/space/rig
+	var/obj/item/clothing/gloves/G = null
+	var/obj/item/clothing/shoes/magboots/MB = null
+	var/obj/item/weapon/tank/T = null
 	var/obj/item/weapon/cell/cell = null
+
+	var/head_type = /obj/item/clothing/head/helmet/space/rig
+	var/boots_type =  null
+	var/gloves_type = null
+	var/tank_type = null
 	var/cell_type = /obj/item/weapon/cell/high //The cell_type we're actually using
-	var/list/modules = list()
-	actions_types = list(/datum/action/item_action/toggle_rig_helmet)
+
+	var/mob/living/carbon/human/wearer
 
 /obj/item/clothing/suit/space/rig/New()
 	..()
-	cell = new cell_type
-	H = new head_type
+	if(cell_type)
+		cell = new cell_type(src)
+	if(head_type)
+		H = new head_type(src)
+		H.canremove = FALSE
+		H.rig = src
+	if(gloves_type)
+		G = new gloves_type(src)
+		G.canremove = FALSE
+	if(boots_type)
+		MB = new boots_type(src)
+		MB.canremove = FALSE
+	if(tank_type)
+		T = new tank_type(src)
+	if(initial_modules && initial_modules.len)
+		for(var/path in initial_modules)
+			var/obj/item/rig_module/new_module = new path(src)
+			modules += new_module
+	if(!(src in processing_objects))
+		processing_objects.Add(src)
 
 /obj/item/clothing/suit/space/rig/Destroy()
-	qdel(cell)
-	cell = null
-	if(H && (H.loc == src || !H.loc))
-		qdel(H)
+	if(processing_objects.Find(src))
+		processing_objects.Remove(src)
+	for(var/obj/item/I in list(H,G,T,MB))
+		if(I && (I.loc == src || !I.loc))
+			qdel(I)
 	H = null
+	G = null
+	T = null
+	MB = null
 	for(var/obj/M in modules)
 		qdel(M)
-	modules.Cut()
+	modules = null
+	if(cell)
+		qdel(cell)
+	cell = null
+	wearer = null
+	if(processing_objects.Find(src))
```

Comment:
`processing_objects -= src` is strictly equivalent and saves you a call to `Find()`

Reply:
i guess